File: Guiões/exemplos/poly/java/Polynomial.java
//package polynomial;

public class Polynomial
{
    private double[] coef = new double[3];
    
    public Polynomial(double[] c)
    {
		coef[0] = c[0];
		coef[1] = c[1];
		coef[2] = c[2];
	}

    public double[] getRoots()
    {
        double s = coef[1] * coef[1] - 4 * coef[0] * coef[2];
    	if (s < 0) 
        {
	        return new double[0];
        }
        else if (s == 0)
        {
			double[] roots = new double[1];
			roots[0] = -coef[1] / (2 * coef[0]);
			return roots;
	    }
	    else
        {
			double[] roots = new double[2];
			roots[0] = (-coef[1] - Math.sqrt (s)) / (2 * coef[0]);
	    	roots[1] = (-coef[1] + Math.sqrt (s)) / (2 * coef[0]);
	    	return roots;
	    }
    }
}

File: Bernardo/exemplos/poly/C/Polynomial.c
#include <math.h>

#include "Polynomial.h"

struct roots getRoots(double coef[3])
{
    double s = coef[1] * coef[1] - 4 * coef[0] * coef[2];
    struct roots r;
    
    if (s < 0) 
    {
		r.length = 0;
		return r;
	}
    else if (s == 0)
    {
		r.length = 1;
		r.root[0] = -coef[1] / (2 * coef[0]);
		return r;
	}
	else
    {
		r.length = 2;
		r.root[0] = (-coef[1] - sqrt (s)) / (2 * coef[0]);
	    r.root[1] = (-coef[1] + sqrt (s)) / (2 * coef[0]);
	    return r;
	}
}


